File: 1800.maximum-ascending-subarray-sum/solution.kt
class Solution {
    fun maxAscendingSum(nums: IntArray): Int {
        return nums.runningFold(0 to 0) { (sum, previousNum), num ->
            if (num <= previousNum) {
                num to num
            } else {
                sum + num to num
            }
        }
        .maxBy { it.first }
        .first
    }
}

File: 1800.maximum-ascending-subarray-sum/solution.go
func maxAscendingSum(nums []int) int {
    sum := nums[0]
    maxSum := sum
    for i := 1; i < len(nums); i++ {
        if nums[i] <= nums[i - 1] {
            sum = nums[i]
        } else {
            sum += nums[i]
        }
        maxSum = max(maxSum, sum)
    }
    return maxSum
}